### PR TITLE
New module item progression

### PR DESCRIPTION
**Pull request by armstrongnate**
refs: MBL-14158
affects: student
release note: none

Test plan:
* Enable the student_modules Experimental Feature
* narmstrong > student5 > Modules course
* Progress through the Modules starting with Prerequisite Module then
move through the items in the Sequential Module
* Module list will display the due date, the completion requirement,
and points possible
* Locked modules will appear disabled in the module list
  - though you can access them via the next/previous buttons
  - Viewing a locked item will display a lock screen
    - Except assignments, they show the assignent details with the lock
    status there
* You must grade the "Must Score" assignment as a Teacher with an 8 or
better in order to move on passed that one
* You can use the kebab in the top right to mark an item as "Done" and "Undone"

### CODE REVIEW

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/zbevn59) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fzbevn59)
teacher: [Link](https://rebrand.ly/hvo5uvw) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fhvo5uvw)
parent: [Link](https://rebrand.ly/uthk94w) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Futhk94w)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-14158](https://instructure.atlassian.net/browse/MBL-14158)

<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/Core/Core/Modules/ModuleItems/ModuleItemDetailsViewController.swift:108:67: Colon Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)
    Linting 'AlertAction.swift' (639/736)
    Linting 'DynamicButton.swift' (640/736)
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Modules/ModuleList/ModuleListViewController.swift [hunk 1]
```diff
@@ -85,13 +85,15 @@ public class ModuleListViewController: UIViewController, ColoredNavViewProtocol
         tableView.registerCell(EmptyCell.self)
         tableView.registerCell(LoadingCell.self)
         tableView.registerHeaderFooterView(ModuleSectionHeaderView.self, fromNib: false)
+        //tableView.registerHeaderFooterView(ModuleSectionFooterView.self, fromNib: false)
```
Comment (bluwave) [inline Core/Core/Modules/ModuleList/ModuleListViewController.swift:88]: Probably don't need to keep this.

Comment (armstrongnate) [reply Core/Core/Modules/ModuleList/ModuleListViewController.swift]: doh thanks

**Comment by armstrongnate** [comment]
@bluwave can you try this again with `student7`? I think the student5 and student6 might not have worked because their enrollments were pending.

**Pull request merged**